Lib/multiprocessing/popen_fork.py

```diff
@@ -66,15 +66,18 @@ def kill(self):
     def _launch(self, process_obj):
         code = 1
         parent_r, child_w = os.pipe()
+        child_r, parent_w = os.pipe()
         self.pid = os.fork()
         if self.pid == 0:
             try:
                 os.close(parent_r)
-                code = process_obj._bootstrap()
+                os.close(parent_w)
+                code = process_obj._bootstrap(parent_sentinel=child_r)
             finally:
                 os._exit(code)
         else:
             os.close(child_w)
+            os.close(child_r)
             self.finalizer = util.Finalize(self, os.close, (parent_r,))
```

Comment:
Same remark as in `popen_spawn_posix.py`: we need to add a finalizer that closes `parent_w`.